projects/plugins/jetpack/extensions/blocks/subscriptions/view.js

```diff
@@ -14,33 +14,33 @@ domReady( function () {
 	if ( ! form.payments_attached ) {
 		form.payments_attached = true;
 		form.addEventListener( 'submit', function ( event ) {
+			event.preventDefault();
+
 			const email = form.querySelector( 'input[type=email]' ).value;
-			const email_clause = email ? `&email=${ encodeURIComponent( email ) }` : '';
-			const post_id = form.querySelector( 'input[name=post_id]' )?.value;
-			const post_id_clause = post_id ? `&post_id=${ post_id }` : '';
-			const tier_id = form.querySelector( 'input[name=tier_id]' )?.value;
-			const tier_id_clause = tier_id ? `&tier_id=${ tier_id }` : '';
 
 			if ( form.resubmitted || ! email ) {
 				return;
 			}
-			event.preventDefault();
 
-			const url =
-				'https://subscribe.wordpress.com/memberships/?' +
-				'blog=' +
-				form.dataset.blog +
-				'&plan=newsletter' +
-				'&source=jetpack_subscribe' +
-				'&post_access_level=' +
-				form.dataset.post_access_level +
-				'&display=alternate' +
-				post_id_clause +
-				tier_id_clause +
-				email_clause;
+			const post_id = form.querySelector( 'input[name=post_id]' )?.value ?? '';
+			const tier_id = form.querySelector( 'input[name=tier_id]' )?.value ?? '';
+
+			const params = new URLSearchParams( {
+				email: encodeURIComponent( email ),
+				post_id,
+				tier_id,
+				blog: form.dataset.blog,
+				plan: 'newsletter',
+				source: 'jetpack_subscribe',
+				post_access_level: form.dataset.post_access_level,
+				display: 'alternate',
+				TB_iframe: true,
+			} );
+
+			const url = 'https://subscribe.wordpress.com/memberships/?' + params.toString();
 
 			window.scrollTo( 0, 0 );
-			tb_show( null, url + '&TB_iframe=true', null );
```

Comment:
That is used by thickBox and probably should not be removed...

Reply:
NVM. I see it was added to the iframe URL...
I would keep it outside so we know it is only for thickbox...

Reply:
I can add a note on the line to make it clear; I find it easier to keep things together in one spot vs previously URL was split into 3 blocks of code. What do you think?

Reply:
Don't get me wrong. This refactor is way nicer.
But I'd personaly keep `url + '&TB_iframe=true'` here as it is very specific for thickbox... I prefer isolating concepts.

As we will end up removing thickbox soon, I can bet that  `TB_iframe=true` will be overlooked and kept in the URL if we add your change...

Reply:
sure 👍